src/warc2zim/content_rewriting/html.py

```diff
@@ -119,25 +91,31 @@ def handle_starttag(self, tag: str, attrs: AttrsList, *, auto_close: bool = Fals
             self.send(self.pre_head_insert)
 
     def handle_endtag(self, tag: str):
-        self._active_tag = None
+        self.html_rewrite_context = None
         if tag == "head" and self.post_head_insert:
             self.send(self.post_head_insert)
         self.send(f"</{tag}>")
 
     def handle_startendtag(self, tag: str, attrs: AttrsList):
         self.handle_starttag(tag, attrs, auto_close=True)
-        self._active_tag = None
+        self.html_rewrite_context = None
 
     def handle_data(self, data: str):
-        if self._active_tag == "title" and self.title is None:
+        if self.html_rewrite_context == "title" and self.title is None:
             self.title = data.strip()
-        elif self._active_tag == "style":
+        elif self.html_rewrite_context == "style":
             data = self.css_rewriter.rewrite(data)
-        elif self._active_tag == "script":
-            rules = get_ds_rules(self.url_rewriter.article_url.value)
+        elif self.html_rewrite_context and self.html_rewrite_context.startswith("js-"):
             if data.strip():
-                data = JsRewriter(self.url_rewriter, rules).rewrite(data)
-        elif self._active_tag == "json":
+                data = JsRewriter(
+                    url_rewriter=self.url_rewriter,
+                    extra_rules=get_ds_rules(self.url_rewriter.article_url.value),
+                    notify_js_module=self.notify_js_module,
+                ).rewrite(
+                    data,
+                    opts={"isModule": self.html_rewrite_context == "js-module"},
+                )
+        elif self.html_rewrite_context == "json":
             if data.strip():
                 rules = get_ds_rules(self.url_rewriter.article_url.value)
                 if rules:
```

Comment:
I would keep the simple `self.html_rewrite_context == "foo"` test together.

Reply:
You mean, push the `self.html_rewrite_context and self.html_rewrite_context.startswith("js-")` to the end?

Reply:
yes